src/cdomain/value/cdomains/mutexAttrDomain.ml

```diff
@@ -21,7 +22,7 @@ end
 include Lattice.FlatConf (struct include Printable.DefaultConf let bot_name = "Uninitialized" let top_name = "Top" end) (MutexKind)
 
 (* Needed because OS X is weird and assigns different constants than normal systems... :( *)
-let recursive_int = lazy (
+let recursive_int = DomainsafeLazy.from_fun (fun () ->
```

Comment:
Why is `DomainsafeLazy` only used here? There are other top-level `lazy` values like
* `Offset.Index.Exp.all`
* `Cilfacade.any_index_exp`
* `LibraryFunctions.intmax_t`

There's also `ResettableLazy` which is used all over. Are all of those safe then?

Reply:
Good point! Also what about `LazyEval`? Are we fine with force being called concurrently there?

Reply:
These two were the ones discovered by Felix back then.

Currently it is quite out of scope to go through all analysis and decide for each if they are thread-safe. In order to keep this moving (this branch does not have the solvers and they are slowly diverging) we can:
a. remove this for now, continue with the introduction of `create` nodes and re-add this once the solvers are in. Then we can test more carefully to see if this really is necessary and why.
b.  Mark this with a comment, make an issue and make a more systematic analysis of the whole situation later.

Reply:
I would be OK with leaving this as the only case right now, but having a new issue to list the possible other places we've listed above.